src/FluentValidation.AspNetCore/FluentValidationMvcConfiguration.cs

```diff
@@ -68,8 +45,7 @@ public class FluentValidationMvcConfiguration {
 		internal bool ClientsideEnabled = true;
 		internal Action<FluentValidationClientModelValidatorProvider> ClientsideConfig = x => {};
 		internal List<Assembly> AssembliesToRegister { get; } = new List<Assembly>();

```

Comment:
If we're going to move the TypeFilter and ServiceLifetime to the base class, then I feel like the `AssembliesToRegister` property, along with the various `RegisterValidatorsFromAssembly` methods should live on the base class too, but this presents an issue with the return typing. It may be possible to solve this by introducing a type parameter which indicates the return type for the fluent configuration methods.